functions/kubernetes/k8sCommand.js

```diff
@@ -1,21 +1,56 @@
 // Runs a job as a Pod (Kubernetes Job) in a Kubernetes cluster
 
 const k8s = require('@kubernetes/client-node');
+var BufferManager = require('./buffer_manager.js').BufferManager;
 var submitK8sJob = require('./k8sJobSubmit.js').submitK8sJob;
 var fs = require('fs');
 
-async function k8sCommand(ins, outs, context, cb) {
+// Function k8sCommandGroup
+//
+// Inputs:
+// - bufferItems - array containing objects with following properties:
+//   * ins
+//   * outs
+//   * context
+//   * cb
+async function k8sCommandGroup(bufferItems) {
 
   let startTime = Date.now();
-  console.log("k8sCommand started, time:", startTime);
+  console.log("k8sCommandGroup started, time:", startTime);
+
+  // Extract particular arrays from buffer items
+  let jobArr = [];
+  let taskIdArr = [];
+  let contextArr = [];
+  let cbArr = [];
+  for (let i=0; i<bufferItems.length; i++) {
+    let bufferItem = bufferItems[i];
+    let ins = bufferItem.ins;
+    let outs = bufferItem.outs;
+    let context = bufferItem.context;
+    let cb = bufferItem.cb;
+
+    var job = context.executor; // object containing 'executable', 'args' and others
+    job.name = context.name;
+    job.ins = ins;
+    job.outs = outs;
+
+    jobArr.push(job);
+    taskIdArr.push(context.taskId);
+    contextArr.push(context);
+    cbArr.push(cb);
+  }
+
+  let context = contextArr[0];
```

Comment:
I understand [0] is rather arbitrary?

Reply:
In _k8sCommand.js_ `contextArr[0]` is used only for partitioning (cloud bursting), but I am not sure how it should behave in case of agglomerations - maybe it should be disabled?